lib/gitlib.sh

```diff
@@ -201,3 +201,24 @@ gitlib::push_master () {
   logecho -n "Pushing$dryrun_flag master branch: "
   logrun -s git push$dryrun_flag origin master || return 1
 }
+
+##############################################################################
+# Ensure running with a synced repo.
+gitlib::repo_state () {
+  # Find remote
+  local remote=$(git -C $TOOL_ROOT remote -v |\
+                 awk '/kubernetes\/release \(fetch\)/ {print $1}')
+  local output=$(git -C $TOOL_ROOT fetch --dry-run $remote)
```

Comment:
hm, this alone won't tell you if the checked-out branch is up-to-date. i.e. I can checkout `master`, wait a week, `git fetch upstream`, and this check will be happy, even though `master` is out-of-date.